src/main/scala/uk/gov/nationalarchives/OcflService.scala

```diff
@@ -47,18 +48,43 @@ class OcflService(ocflRepository: OcflRepository) {
   def filterChangedObjects(objects: List[DisasterRecoveryObject]): List[DisasterRecoveryObject] =
     objects.filter(obj => isChecksumMismatched(obj.id, obj.name, obj.checksum))
 
-  private def isChecksumMismatched(ioId: UUID, fileName: String, checksum: String): Boolean =
+  def getMissingAndChangedObjects(
+      objects: List[DisasterRecoveryObject]
+  ): (List[DisasterRecoveryObject], List[DisasterRecoveryObject]) = {
+    val missingAndNonMissingObjects: Map[String, List[DisasterRecoveryObject]] =
+      objects.foldLeft(Map[String, List[DisasterRecoveryObject]]("missingObjects" -> Nil, "changedObjects" -> Nil)) {
+        case (objectMap, obj) =>
+          val objectId = obj.id
+          val potentialOcflObject = Try(ocflRepository.getObject(objectId.toHeadVersion))
+          lazy val missedObjects = objectMap("missingObjects")
+          lazy val changedObjects = objectMap("changedObjects")
+
+          potentialOcflObject match {
+            case Success(ocflObject) =>
+              val checksumUnchanged =
+                Option(ocflObject.getFile(s"$objectId/${obj.name}"))
+                  .map(ocflFileObject => ocflFileObject.getFixity.get(DigestAlgorithm.sha256))
+                  .contains(obj.checksum)
+              if (checksumUnchanged) objectMap else objectMap + ("changedObjects" -> (obj :: changedObjects))
+
+            case Failure(_) => objectMap + ("missingObjects" -> (obj :: missedObjects))
+          }
+      }
+
+    (missingAndNonMissingObjects("missingObjects"), missingAndNonMissingObjects("changedObjects"))
+  }
+
+  private def isChecksumMismatched(objectId: UUID, fileName: String, checksum: String): Boolean =
     Option
-      .when(ocflRepository.containsObject(ioId.toString)) {
-        Option(ocflRepository.getObject(ioId.toHeadVersion).getFile(s"$ioId/$fileName"))
-          .map(a => a.getFixity.get(DigestAlgorithm.sha256))
+      .when(ocflRepository.containsObject(objectId.toString)) {
+        Option(ocflRepository.getObject(objectId.toHeadVersion).getFile(s"$objectId/$fileName"))
+          .map(ocflFileObject => ocflFileObject.getFixity.get(DigestAlgorithm.sha256))
```

Comment:
You can replace `ocflFileObject` with `_` as well here.

Reply:
I just realised that this method is no longer needed